src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,191 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private Node<T> currentNode;
+    private Node<T> previousNode;
+    private Node<T> nextNode;
+    private int size;
+
+    public static class Node<T> {
+        private T item;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(Node<T> prev, T item, Node<T> next) {
+            this.prev = prev;
+            this.next = next;
+            this.item = item;
+        }
+    }
```

Comment:
The Node class should be an inner class and placed at the end of the MyLinkedList class, as per the checklist requirement: 'Make Node<T> an inner class.'